tool-openssl/ecparam.cc

```diff
@@ -2,69 +2,75 @@
 // SPDX-License-Identifier: Apache-2.0 OR ISC
 
 #include <openssl/ec.h>
-#include <openssl/pem.h>
 #include <openssl/err.h>
 #include <openssl/obj.h>
+#include <openssl/pem.h>
 #include "../tool/internal.h"
 #include "internal.h"
 
-enum OutputFormat {
-  FORMAT_PEM = 1,
-  FORMAT_DER = 2
-};
+enum OutputFormat { FORMAT_PEM = 1, FORMAT_DER = 2 };
 
-static bssl::UniquePtr<EC_GROUP> ValidateAndCreateECGroup(const std::string& curve_name) {
-    int nid = OBJ_sn2nid(curve_name.c_str());
-    if (nid == NID_undef) {
-        nid = EC_curve_nist2nid(curve_name.c_str());
-    }
+static bssl::UniquePtr<EC_GROUP> ValidateAndCreateECGroup(
+    const std::string &curve_name) {
+  int nid = OBJ_sn2nid(curve_name.c_str());
+  if (nid == NID_undef) {
+    nid = EC_curve_nist2nid(curve_name.c_str());
+  }
 
-    bssl::UniquePtr<EC_GROUP> group;
-    if (nid != NID_undef) {
-        group.reset(EC_GROUP_new_by_curve_name(nid));
-    }
+  // We only allow 1 short name per NID but P-256 are synonymously addressible
+  // by secp256r1 and prime256v1
+  if (nid == NID_undef && isStringUpperCaseEqual(curve_name, "secp256r1")) {
+    nid = NID_X9_62_prime256v1;
+  }
+
```

Comment:
I think a better solution is to add it [here](https://github.com/aws/aws-lc/blob/main/crypto/obj/obj_dat.h#L9803) to our list of short-names:
```
❯ git diff
diff --git a/crypto/obj/obj_dat.h b/crypto/obj/obj_dat.h
index 6834ec795..d74f7ed0c 100644
--- a/crypto/obj/obj_dat.h
+++ b/crypto/obj/obj_dat.h
@@ -9800,6 +9800,7 @@ static const uint16_t kNIDsInShortNameOrder[] = {
     712 /* secp224k1 */,
     713 /* secp224r1 */,
     714 /* secp256k1 */,
+    415 /* secp256r1 */
     715 /* secp384r1 */,
     716 /* secp521r1 */,
     154 /* secretBag */,
```

With this, this name would resolve in other contexts as well.